internal/controller/util.go

```diff
@@ -27,13 +27,15 @@ import (
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
 const (
-	PhaseResourcesCreated string = "resources-created"
-
-	PhaseReady string = "ready"
+	PhaseResourcesCreated string = "ResourcesCreated"
+	PhaseReady            string = "Ready"
```

Comment:
It will break backward-compatibility. The changes looks good. But You need to make sure the existing deployments with old phase statuses won't get affected. The new operator version won't recognize the old phases. Please open a separate PR with backward-compatible changes.